app/lib/frontend/handlers.dart

```diff
@@ -221,36 +219,29 @@ Future<shelf.Response> packagesHandler(shelf.Request request) async {
   }
 }
 
-/// Handles requests for /static
-final StaticsCache staticsCache = new StaticsCache();
-
 /// The max age a browser would take hold of the static files before checking
 /// with the server for a newer version.
 const _staticMaxAge = const Duration(minutes: 5);
 
 Future<shelf.Response> staticsHandler(shelf.Request request) async {
   // Simplifies all of '.', '..', '//'!
   final String normalized = path.normalize(request.requestedUri.path);
-  if (normalized.startsWith('/static/')) {
-    final assetPath = '$_staticPath/${normalized.substring('/static/'.length)}';
-
-    final StaticFile staticFile = staticsCache.staticFiles[assetPath];
-    if (staticFile != null) {
-      final ifModifiedSince = request.ifModifiedSince;
-      if (ifModifiedSince != null &&
-          !staticFile.lastModified.isAfter(ifModifiedSince)) {
-        return new shelf.Response.notModified();
-      }
-      return new shelf.Response.ok(
-        staticFile.bytes,
-        headers: {
-          HttpHeaders.CONTENT_TYPE: staticFile.contentType,
-          HttpHeaders.CONTENT_LENGTH: staticFile.bytes.length.toString(),
-          HttpHeaders.LAST_MODIFIED: formatHttpDate(staticFile.lastModified),
-          HttpHeaders.CACHE_CONTROL: 'max-age: ${_staticMaxAge.inSeconds}',
-        },
-      );
+  final StaticFile staticFile = staticsCache.getFile(normalized);
+  if (staticFile != null) {
+    final ifModifiedSince = request.ifModifiedSince;
+    if (ifModifiedSince != null &&
+        !staticFile.lastModified.isAfter(ifModifiedSince)) {
+      return new shelf.Response.notModified();
     }
+    return new shelf.Response.ok(
+      staticFile.bytes,
+      headers: {
+        HttpHeaders.CONTENT_TYPE: staticFile.contentType,
+        HttpHeaders.CONTENT_LENGTH: staticFile.bytes.length.toString(),
+        HttpHeaders.LAST_MODIFIED: formatHttpDate(staticFile.lastModified),
```

Comment:
If we're going through all of this work, we should switch to etags based on file hashes...

make it a todo for now, I guess

Reply:
`ETag` over the currently in-use `If-Modified-Since` does improve a bit, but not too much. Added TODOs though, because it will be easy with the hashed content.